File: v2/python/anhurdb/models/enums.py
from enum import Enum

class MemoryType(str, Enum):
    """
    Cognitive memory types defined by the AnhurDB epistemology.
    """
    # Canonical AnhurDB taxonomy (13 types).
    #
    # Junior Tip [taxonomy SSOT]: the authority is AnhurCore's core.yaml, mirrored
    # by the server's schema.MemoryTypes and the MCP list_types tool — not this
    # file. list_types() below derives straight from this enum, so a missing
    # member silently shortens the public taxonomy. Keep Go, Python, TypeScript
    # and core.yaml in step, in the same change.
    EPISODIC = "episodic"
    FACT = "fact"
    PREFERENCE = "preference"
    DECISION = "decision"
    TASK = "task"
    RISK = "risk"
    REASONING = "reasoning"
    IDEA = "idea"
    EMOTION = "emotion"
    CONSOLIDATED = "consolidated"
    HUB = "hub"
    FILE = "file"
    ROUTER = "router"  # Macro-theme backbone: a hub of hubs (ADR-0005)

class MemoryStatus(str, Enum):
    """
    Processing status of a memory record.
    """
    SAVED = "saved"                                # Written, awaiting processing
    PENDING = "pending"                            # Queued for processing
    PROCESSING = "processing"                      # Being processed
    COMPLETED = "completed"                        # Processing complete
    FAILED = "failed"                              # Processing failed
    ARCHIVED = "archived"                          # Soft-deleted
    DECAYED = "decayed"                            # Low-fidelity (memory decay applied)
    CONSOLIDATED = "consolidated"                  # Included in a summary record
    LINKED = "linked"                              # Cross-session links established
    HUBBED = "hubbed"                              # Grouped into a topic cluster
    PENDING_JUDGE = "pending_judge"                # Pending review
    FAILED_JUDGE = "failed_judge"                  # Review rejected
    FAILED_CONSOLIDATION = "failed_consolidation"  # Summarization failed

File: v2/golang/models/enums.go
/*
Package models defines the cognitive memory types and statuses for AnhurDB.

These constants match the canonical AnhurDB taxonomy; all three SDKs share
the same values.
*/
package models

// MemoryType defines the cognitive memory types in AnhurDB.
// Canonical AnhurDB taxonomy (13 types).
//
// Junior Tip [taxonomy SSOT]: this list is NOT free-form. Its authority is
// AnhurCore's core.yaml, mirrored by the server's schema.MemoryTypes and by the
// MCP list_types tool. A type declared here but absent server-side is rejected
// with HTTP 400 on create; a type served by the server but missing here silently
// shrinks ListTypes() and — in TypeScript — makes the value a compile error for
// the caller. Keep all three SDKs and core.yaml in step, in the same change.
type MemoryType string

const (
	TypeEpisodic     MemoryType = "episodic"     // Raw conversation turns
	TypeFact         MemoryType = "fact"          // Verifiable information
	TypePreference   MemoryType = "preference"    // User likes/dislikes
	TypeDecision     MemoryType = "decision"      // Team or personal choices
	TypeTask         MemoryType = "task"          // Action items
	TypeRisk         MemoryType = "risk"          // Concerns and warnings
	TypeReasoning    MemoryType = "reasoning"     // Chain of thought
	TypeIdea         MemoryType = "idea"          // Proposals
	TypeEmotion      MemoryType = "emotion"       // Expressed feelings
	TypeConsolidated MemoryType = "consolidated"  // Agent-synthesised summary
	TypeHub          MemoryType = "hub"           // Cross-session cluster
	TypeFile         MemoryType = "file"          // Uploaded document root
	TypeRouter       MemoryType = "router"        // Macro-theme backbone: a hub of hubs (ADR-0005)
)

// MemoryStatus defines the lifecycle status of a record.
type MemoryStatus string

const (
	StatusSaved              MemoryStatus = "saved"               // Default on creation
	StatusPending            MemoryStatus = "pending"             // Legacy (pre-v2)
	StatusConsolidated       MemoryStatus = "consolidated"        // After consolidation
	StatusArchived           MemoryStatus = "archived"            // Soft-deleted
	StatusDecayed            MemoryStatus = "decayed"             // Memory decay applied
	StatusProcessing         MemoryStatus = "processing"          // Being ingested
	StatusCompleted          MemoryStatus = "completed"           // Ingestion done
	StatusLinked             MemoryStatus = "linked"              // Similarity edges set
	StatusHubbed             MemoryStatus = "hubbed"              // Assigned to hub node
	StatusFailed             MemoryStatus = "failed"              // Generic failure
	StatusPendingJudge       MemoryStatus = "pending_judge"       // Awaiting Judge
	StatusFailedJudge        MemoryStatus = "failed_judge"        // Judge rejected
	StatusFailedConsolidation MemoryStatus = "failed_consolidation" // Consolidation failed
)
